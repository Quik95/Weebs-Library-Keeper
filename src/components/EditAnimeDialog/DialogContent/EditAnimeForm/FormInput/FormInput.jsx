import React from "react";
import PropTypes from "prop-types";

// MUI
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";

//styles
import useStyles from "./FormInput.styles";

const FormInput = function FormInput({ title, id, type = "text", value }) {
  const classes = useStyles();
  const error = false;

  return (
    <Grid item xs={12} sm={6} md={4} className={classes.grid}>
      <TextField
        error={error}
        id={id}
        label={title}
        // value={value}
        defaultValue={value}
        variant="outlined"
        type={type}
      />
    </Grid>
  );
};

FormInput.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
};

export default FormInput;
