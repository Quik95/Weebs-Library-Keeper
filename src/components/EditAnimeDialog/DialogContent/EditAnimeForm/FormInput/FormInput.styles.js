import {
  makeStyles
} from '@material-ui/core/styles'

export default makeStyles(theme => ({
  grid: {
    display: 'flex',
    justifyContent: 'center'
  },
  [theme.breakpoints.down('xs')]: {
    formControl: {
      width: '100%'
    }
  }
}))