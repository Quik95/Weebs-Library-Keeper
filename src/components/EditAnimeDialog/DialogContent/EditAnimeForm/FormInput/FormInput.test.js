import FormInput from './FormInput';

describe('<FormInput/>', () => {
  const wrapper = shallow(<FormInput title="Dr. Stone" id="title" />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
