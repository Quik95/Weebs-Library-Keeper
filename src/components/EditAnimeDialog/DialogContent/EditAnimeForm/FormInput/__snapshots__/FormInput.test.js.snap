// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput/> Renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={4}
>
  <WithStyles(ForwardRef(FormControl))
    error={false}
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      htmlFor="title"
    >
      Edit Dr. Stone
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Input))
      aria-describedby="edit-anime-Dr. Stone"
      id="title"
      type="text"
      variant="outlined"
    />
  </WithStyles(ForwardRef(FormControl))>
</WithStyles(ForwardRef(Grid))>
`;
